renku/cli/dataset.py

```diff
@@ -157,6 +160,106 @@ def add(client, name, urls, link, relative_to, target, force):
         raise BadParameter('Could not process {0}'.format(url))
 
 
+@dataset.command('ls-files')
+@click.option(
+    '--dataset', multiple=True, help='Filter files in specific dataset.'
+)
+@click.option(
+    '--authors',
+    help='Filter files which where authored by specific authors. '
+    'Multiple authors are specified by comma.'
+)
+@click.option(
+    '-I',
+    '--include',
+    default=None,
+    multiple=True,
+    help='Include files matching given pattern.'
+)
+@click.option(
+    '-X',
+    '--exclude',
+    default=None,
+    multiple=True,
+    help='Exclude files matching given pattern.'
+)
+@click.option(
+    '--format',
+    type=click.Choice(DATASET_FILES_FORMATS),
+    default='tabular',
+    help='Choose an output format.'
+)
+@pass_local_client(clean=False, commit=False)
+def ls_files(client, format, exclude, include, authors, dataset):
+    """List files in dataset."""
+    records = _filter(
+        client,
+        dataset_names=dataset,
+        authors=authors,
+        include=include,
+        exclude=exclude
+    )
+
+    DATASET_FILES_FORMATS[format](client, records)
+
+
+def _include_exclude(file_path, include=None, exclude=None):
+    """Check if file matches one of include filters and not in exclude filter.
+
+    :param file_path: Path to the file.
+    :param include: Tuple containing patterns to which include from result.
+    :param exclude: Tuple containing patterns to which exclude from result.
+    """
+    if exclude is not None and exclude:
+        for pattern in exclude:
+            if fnmatch(str(file_path), pattern):
```

Comment:
```suggestion
            if file_path.match(pattern):
```

https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.match